src/main/java/com/dev/cinema/service/impl/UserServiceImpl.java

```diff
@@ -1,34 +1,36 @@
 package com.dev.cinema.service.impl;
 
+import com.dev.cinema.config.SecurityConfig;
 import com.dev.cinema.dao.UserDao;
 import com.dev.cinema.model.User;
 import com.dev.cinema.service.UserService;
-import com.dev.cinema.util.HashUtil;
-import java.util.Optional;
 import org.springframework.stereotype.Service;
 
 @Service
 public class UserServiceImpl implements UserService {
     private final UserDao userDao;
+    private final SecurityConfig securityConfig;
 
-    public UserServiceImpl(UserDao userDao) {
+    public UserServiceImpl(UserDao userDao, SecurityConfig securityConfig) {
         this.userDao = userDao;
+        this.securityConfig = securityConfig;
     }
 
     @Override
     public User add(User user) {
-        user.setSalt(HashUtil.getSalt());
-        user.setPassword(HashUtil.hashPassword(user.getPassword(), user.getSalt()));
+        user.setPassword(securityConfig.getEncoder().encode(user.getPassword()));
         return userDao.add(user);
     }
 
     @Override
-    public Optional<User> findByEmail(String email) {
-        return userDao.findByEmail(email);
+    public User findByEmail(String email) {
+        return userDao.findByEmail(email).orElseThrow(() ->
+                new RuntimeException("Can't get by email " + email));
```

Comment:
feel free to throw NoSuchElementException and handle this exception in the GlobalExceptionHandler (second part is not required in this HW)